social-tw/social-tw-website
# [UST-140] [Frontend] feat: forbid user actions

## Summary
- [x] Add `UI` :: `ForbidActionDialog`
- [x] Run `API` :: `FetchUserReputation`
- [x] Update `Hook` :: `CreatePost`
- [x] Update `Hook` :: `CreateComment`
- [x] Update `Hook` :: `RemoveComment`
- [x] Update `Hook` :: `CreateVote`
- [x] Show `UI` :: `ForbidActionDialog` if bad user do any action.

## Linked Issue

close https://github.com/social-tw/social-tw-website/issues/453

## Details

- Use Dialog store to control ReputationDialog visibility.

## Impacted Areas

- CreatePost
- useCreatePost
- useCreateComment
- useRemoveComment
- useVotes
- ForbidActionDialog
- layout
- api

## Tests

- useCreatePost.test
- useVotes.test

## Possible Impacts

- CreatePost
- useCreatePost
- useCreateComment
- useRemoveComment
- useVotes
- ForbidActionDialog
- layout
- api

## Visual Materials

![Screenshot 2024-07-27 at 14 46 26](https://github.com/user-attachments/assets/67733cc0-9f40-4523-8442-fdfb867be400)

## Verification Steps

- When "Create Post, Comment, Vote, Remove Comment", you should see the ForbidActionDialog

## Checklist
